backend/main/views/answer.py

```diff
@@ -1,12 +1,30 @@
 from rest_framework import generics
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
 from ..serializers.answer_serializer import AnswerSerializer
 from ..models.answer import Answer
 
-class AnswerList(generics.ListAPIView):
-    queryset = Answer.objects.all()
-    serializer_class = AnswerSerializer
-    filterset_fields = ['user_id', 'lesson_id']
+class AnswerList(APIView):
+    def get(self, request):
+        answers = Answer.objects.all()
+        serializer = AnswerSerializer(answers, many=True)
+
+        return Response(serializer.data)
+
+    def post(self, request):
+        serializer = AnswerSerializer(data=request.data)
+
+        if serializer.is_valid(raise_exception=True):
+            serializer.save()
+
+            response = Response('Answer added', status=status.HTTP_201_CREATED)
+        else:
+            response =  Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
`serializer.is_valid(raise_exception=True)` is already handles validations. It returns `BadRequest` if nay something wrong inside the serializer. No need to put into if statement. Maybe thats why last time, wala ni work properly imo saving cus wala ni sud sa if statement. 

Try to comment your `answer.save()` inside the serializer daw then do this sa views.py. Try to see if it works ghapon

```suggestion

        serializer.is_valid(raise_exception=True):
        serializer.save()

        return Response('Answer added', status=status.HTTP_201_CREATED)

```

Reply:
The POST request returns `'Answer added'` but it doesn't get added in the db since I commented out `answer.save()`